tests/test_data.py

```diff
@@ -500,3 +533,80 @@ def test_hap_gt_write(self):
 
         # remove the file
         os.remove(str(fname))
+
+
+    
+class TestGenotypesRefAlt:
+    def test_read_ref_alt(self):
+        #simple.vcf
+        gts_ref_alt_read = GenotypesRefAlt(DATADIR.joinpath("simple.vcf"))
+        gts_ref_alt_read.read()
+        expected = np.array(
+            [
+                ("T", "C"),
+                ("A", "G"),
+                ("C", "A"),
+                ("A", "G"),
+            ], dtype = gts_ref_alt_read.variants[["ref", "alt"]].dtype
+        )
+        for i, x in enumerate(expected):
+            assert gts_ref_alt_read.variants[["ref", "alt"]][i] == x
+
+        # example.vcf.gz
+        gts_ref_alt = GenotypesRefAlt(DATADIR.joinpath("example.vcf.gz"))
+        gts_ref_alt.read()
+        expected = np.array(
+            [
+                ("C", "A"),
+                ("T", "C"),
+                ("G", "A"),
+                ("T", "C"),
+                ("A", "G"),
+                ("A", "G"),
+                ("T", "G"),
+                ("T", "A"),
+                ("G", "A"),
+                ("T", "C"),
+                ("C", "G"),
+                ("A", "G"),
+                ("T", "C"),
+                ("T", "C"),
+                ("C", "T"),
+            ], dtype = gts_ref_alt.variants[["ref", "alt"]].dtype
+        )
+        for i, x in enumerate(expected):
+            assert gts_ref_alt.variants[["ref", "alt"]][i] == x
+        
+
+    
+    def test_write_ref_alt(self):
+        gts_ref_alt_write = GenotypesRefAlt(DATADIR.joinpath("simple.vcf"))
+        
+        expected = gts_ref_alt_write.read()
+        fname = DATADIR.joinpath("test.vcf")
+        gts_ref_alt_write.fname = fname
+        gts_ref_alt_write.write()
+        assert gts_ref_alt_write.read() == expected
```

Comment:
see my comment above about checking the `data`, `variants`, and `samples` properties here